validator_worker/src/sentry_interface.rs

```diff
@@ -137,30 +154,39 @@ impl<T: Adapter + 'static> SentryApi<T> {
     pub async fn get_event_aggregates(
         &self,
         after: DateTime<Utc>,
-    ) -> Result<EventAggregateResponse, reqwest::Error> {
-        let whoami = self.adapter.whoami();
+    ) -> Result<EventAggregateResponse, Box<ValidatorWorker>> {
+        let mut adapter = self
+            .adapter
+            .write()
+            .expect("get_event_aggregates: Failed to acquire adapter");
+
+        let whoami = adapter.whoami();
+
         let validator = self
             .channel
             .spec
             .validators
             .into_iter()
             .find(|&v| v.id == whoami);
-        let auth_token = self
-            .adapter
-            .get_auth(validator.unwrap())
-            .expect("Failed to get user auth token");
+
+        let auth_token = adapter.get_auth(validator.unwrap());
```

Comment:
Even though we have fairly small amount of code, I see the confusion that leads to duplicating code by accident.

Here is what I don't understand: Why do we even bother to do this check, if we already have it in the initialization of the SentryInterface?
Why search for the `adapter.whoami()` in the `channle.spec.validators` and then even `validator.unwrap()` the value?
This whole thing can be simplified to just:
```
let auth_token = adapter.get_auth(adapter.whoami());
```
If you follow the implementation, the `adapter.whoami()` will always be included in the Channel validators, but it's just not obvious.

Reply:
`adapter.whoami()` returns a `string`,
`.find(|&v| v.id == whoami);` returns a `ValidatorDesc`

Reply:
Also, the idea behind the unwrap() is because we already check it in `new()` so it would always be there